facade/src/main/scala/reactST/reactTable/HTMLTableBuilder.scala

```diff
@@ -0,0 +1,274 @@
+   * parameter. This header will make the maker sortable by clicking
+   * on the header. Will only compile if the column instance is a subtype
+   * of UseSortByColumnPros.
+   *
+   * @param cellClass An optional CSS class to apply to the <th>.
+   * @param useDiv True to use a <div> instead of a <th>. Needed for tables withBlockLayout.
+   * @return
+   */
+  def sortableHeaderCellFn(
+    cellClass: Css = Css.Empty,
+    useDiv:    Boolean = false
+  ): ColumnObject[_] with UseSortByColumnProps[_] => TagMod =
+    col => {
+      def sortIndicator(col: UseSortByColumnProps[_]): TagMod =
+        if (col.isSorted) {
+          val index   = if (col.sortedIndex > 0) (col.sortedIndex + 1).toString else ""
+          val ascDesc = if (col.isSortedDesc.getOrElse(false)) "\u2191" else "\u2193"
+          <.span(s" $index$ascDesc")
+        } else TagMod.empty
+
+      val headerProps = col.getHeaderProps()
+      val toggleProps = col.getSortByToggleProps()
+
+      // if, for example, the table also has block layout, both properties objs will have styles.
+      val styles = combineStyles(headerProps, toggleProps)
+
+      headerCell(useDiv)(
+        TableMaker.props2Attrs(headerProps),
+        TableMaker.props2Attrs(toggleProps),
+        styles,
+        cellClass,
+        col.renderHeader,
+        <.span(sortIndicator(col))
+      )
+    }
+
+  /**
+   * A function to use in TableMaker.makeTable for the
+   * footerCellFn parameter. This is a a basic footer.
+   *
+   * @param cellClass An optional CSS class to apply the the <th>.
+   * @param useDiv True to use a <div> instead of a <th>. Needed for tables withBlockLayout.
+   */
+  def basicFooterCellFn(
+    cellClass: Css = Css.Empty,
+    useDiv:    Boolean = false
+  ): ColumnObject[_] => TagMod = { col =>
+    col.Footer.map(_ =>
+      headerCell(useDiv)(
+        TableMaker.props2Attrs(col.getFooterProps()),
+        cellClass,
+        col.renderFooter
+      )
+    )
+  }
+
+  private def headerCell(useDiv: Boolean) = if (useDiv) <.div(^.className := "th") else <.th()
+
+  /**
+   * Merge the styles from properties objects returned by react-table.
+   * If we just use the properties objects directly, and more than one have "style",
+   * only the last one gets used. Apparently, scalajs-react doesn't merge them. So, we have to combine them manually.
+   * In the case of identical style keys, the last one wins. This is probably why scalajs-react doesn't merge.
+   *
+   * Styles are only combined if more than one propertyObjs have styles. Otherwise, the props2Attrs
+   * method works fine and combining is a waste of time. If one or fewer propertyObjs have styles,
+   * TagMod.empty is returned. See sortableHeaderCellFn for example usage.
+   *
+   * @param propertyObs A varargs of "properties objects" from react-table.
+   * @return A TagMod.
+   */
+  def combineStyles(propertyObs: js.Object*): TagMod = {
+    val styles = propertyObs.collect {
```

Comment:
Should these go in TableMaker? It seems like any renderer could need to combine the styles if they are combining some of the plugins.

Reply:
Ok, did that. Changed things a bit to leverage the fact that `scalajs-react` does have the concept of styles separate from attributes. Also, moved this to a `util` object, since this logic is independent of this facade (and part of it won't be necessary once https://github.com/japgolly/scalajs-react/pull/843 is merged).